crates/telio-sockets/src/protector/apple.rs

```diff
@@ -55,35 +57,33 @@ pub(crate) fn bind_to_tun(sock: NativeSocket, tunnel_interface: u64) -> io::Resu
 }
 
 pub(crate) fn bind(interface_index: u32, socket: i32) -> io::Result<()> {
+    let nz_index = std::num::NonZeroU32::new(interface_index);
+    if nz_index.is_none() {
+        return Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            "Invalid interface index",
+        ));
+    }
+
     if let Some(sock_addr) = getsockname::<SockaddrStorage>(socket)?.family() {
-        let (option, level) = match sock_addr {
-            AddressFamily::Inet6 => (libc::IPV6_BOUND_IF, libc::IPPROTO_IPV6),
-            AddressFamily::Inet => (libc::IP_BOUND_IF, libc::IPPROTO_IP),
+        match sock_addr {
+            AddressFamily::Inet => unsafe {
+                Socket::from_raw_fd(socket).bind_device_by_index_v4(nz_index)?;
```

Comment:
was this tested on macos/ios? like mesh+vpn case. This api looks kinda nice but would be nice to not introduce any bugs

Reply:
Yeah, this was done explicitly due to the failing mesh +vpn tests (although, it wasn't a root cause, but I left it, because of nicer API :smile: ).